body_in.go

```diff
@@ -383,8 +382,14 @@ func compareTotals(inv *bill.Invoice, doc *GeneralDocumentData) error {
 			return err
 		}
 
-		if fatturapaTotal.Compare(inv.Totals.Payable) != 0 {
-			return errors.New("totals do not match")
+		// Calculate to get totals
+		if err = inv.Calculate(); err != nil {
+			return err
+		}
+
+		r := fatturapaTotal.Subtract(inv.Totals.Payable)
```

Comment:
A `nil` check on the `inv.Totals` may be required for extra security.

I'd also try to go with the go way for variable naming:

```suggestion
		r := ft.Subtract(inv.Totals.Payable)
```

Reply:
Makes sense, thanks!